File: src/2869.minimum-operations-to-collect-elements.3044/solution.ts
function minOperations (nums: number[], k: number): number {
    const flags = new Array<boolean>(k + 1);
    let rest = k;
    let index = nums.length - 1;
    while (index > -1) {
        const n = nums[index];
        if (n > k) {
            index--;
            continue;
        }
        if (!flags[n]) {
            flags[n] = true;
            rest--;
            if (rest === 0) {
                return nums.length - index;
            }
        }

        index--;
    }
    return 0;
}

File: src/2869.minimum-operations-to-collect-elements.3044/solution.java
class Solution {
    public int minOperations(List<Integer> nums, int k) {
        Set<Integer> set = new HashSet<>();
        int index = nums.size()-1;
        while (index>-1 && set.size()<k){
            int item = nums.get(index--);
            if(item<=k){
                set.add(item);
                if(set.size() == k){
                    break;
                }
            }
        }
        return nums.size()-index-1;
    }
}